src/Lumper.UI/Updater/Updater.cs

```diff
@@ -0,0 +1,420 @@
+        }
+        else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+        {
+            processInfo = new ProcessStartInfo("cmd.exe", "/c " + command) { CreateNoWindow = true, UseShellExecute = true, RedirectStandardError = true, RedirectStandardOutput = true, WorkingDirectory = Directory.GetCurrentDirectory() };
+
+        }
+        process = Process.Start(processInfo);
+    }
+    /// <summary>
+    /// Grab the JSON from the Github API and deserializes it
+    /// </summary>
+    /// <returns></returns>
+    /// <exception cref="Exception"></exception>
+    private static async Task<GithubUpdate> GetGithubUpdates()
+    {
+        var client = new HttpClient();
+
+        var request = new HttpRequestMessage() {
+            RequestUri = new Uri("https://api.github.com/repos/momentum-mod/lumper/releases"),
+            Method = HttpMethod.Get
+        };
+        client.DefaultRequestHeaders.Add("User-Agent", "Other");
+        HttpResponseMessage response = await client.SendAsync(request);
+
+        if (response.IsSuccessStatusCode)
+        {
+            var jsonString = await response.Content.ReadAsStringAsync();
+
+            return JsonConvert.DeserializeObject<GithubUpdate>(JsonNode.Parse(jsonString)[0].ToString());
+        }
+        else
+        {
+            throw new Exception("Could not connect - error " + response.StatusCode);
+        }
+    }
+    /// <summary>
+    /// Checks for possible update on the Github releases page by the tag name
+    /// </summary>
+    /// <returns>The current latest version, or null if it the latest</returns>
+    public static async Task<Version?> CheckForUpdate()
+    {
+        GithubUpdate assets;
+        try
+        {
+            assets = await GetGithubUpdates();
+        }
+        catch (Exception ex)
+        {
+            ButtonResult result = await MessageBoxManager
+            .GetMessageBoxStandard(
+                "Error",
+                ex.Message, ButtonEnum.Ok)
+            .ShowWindowDialogAsync(Program.Desktop.MainWindow);
+            return null;
+        }
+
+        //parse tag name to find the current and latest version
+        //finding the format of xx.yy.zz
+        Version? current;
+        Version? latest;
+        try
+        {
+            current = GetVersionFromString(Assembly.GetExecutingAssembly().GetName().Version.ToString());
+            latest = GetVersionFromString(assets.TagName);
+        }
+        catch (Exception ex)
+        {
+            ButtonResult result = await MessageBoxManager
+            .GetMessageBoxStandard(
+                "Error",
+                "Could not parse version number.", ButtonEnum.Ok)
+            .ShowWindowDialogAsync(Program.Desktop.MainWindow);
+            return null;
+
+        }
+        if (current != latest)
+            return latest;
+        else
+            return null;
```

Comment:
These `Version`s are only ever going to be null if they *throw*, and if they do, we don't have to catch that here.

Like I said in the comments above, exceptions completely halt the execution of the current method, and jumps immediately to the first `catch` in the call stack. Since we're fine with catching all of these exceptions in one place (in `Update() somewhere, I'll point out below), it's fine to throw anywhere in these methods if something bad happens, then continue on in the below code with confidence that any condition that would have throw if true, is false. For example in the above regex stuff, when we do

```cs
if (!match.Success)
    throw new InvalidDataException("Could not parse version number");

GroupCollection currentVersion = match.Groups;

int.TryParse(currentVersion[1].ToString(), out var major);
 ... etc ...
 ```
`match.Success` is *guaranteed* to be true from the `GroupCollection currentVersion ... etc` line onwards.

So in this block, all we need is the following:
```cs
Version current = GetVersionFromString(Assembly.GetExecutingAssembly().GetName().Version.ToString());;
Version latest = GetVersionFromString(assets.TagName);

if (current != latest)
    return latest;
else
    return null;
```

An upshot of this approach is that return null now simply means that we have an up-to-date version, which is a completely standard thing we expect to happen all the time. If we returned null for parsing errors, we no longer know whether a return value of null is the result of regular, well-behaved behaviour (e.g. program is already up-to-date), or the result of an unintended parsing error (e.g. Github changed their API or something).